gprofiler/profiler_state.py

```diff
@@ -8,22 +8,15 @@
 class ProfilerState:
     # Class for storing generic state parameters. These parameters are the same for each profiler.
     # Thanks to that class adding new state parameters to profilers won't result changing code in every profiler.
-    def __init__(
-        self,
-        stop_event: Event,
-        storage_dir: str,
-        profile_spawned_processes: bool,
-        insert_dso_name: bool,
-        profiling_mode: str,
-        container_names_client: Optional[ContainerNamesClient],
-    ) -> None:
-        self._stop_event = stop_event
-        self._profile_spawned_processes = profile_spawned_processes
-        self._insert_dso_name = insert_dso_name
-        self._profiling_mode = profiling_mode
-        self._temporary_dir = TemporaryDirectoryWithMode(dir=storage_dir, mode=0o755)
+    def __init__(self, **kwargs: Any) -> None:
```

Comment:
Instead of using kwargs (which prevents using types for arguments and is more error prone because unwanted names can be given), what I meant here:

> Can you change the c'tor of ProfilerState to be kwargs-only? (i.e __init__(self, *, ...)) and use kwargs here? This is more readable.

is to use [keyword-only arguments](https://python-3-for-scientists.readthedocs.io/en/latest/python3_advanced.html) , so:

```
def __init__(self, *, stop_event: Event, storage_dir: str, ....)
```

Reply:
I'm sorry for the confusion, should've written kwargs-*only* explicitly haha